autowarefoundation/autoware.universe
# feat(mission_planner)!: support ad api

Signed-off-by: Takagi, Isamu <isamu.takagi@tier4.jp>

## Description

Introduce AD API for routing. This PR is the mission planner part of the diagram.
Here are the changes for each package:
- component_interface_specs: Add the internal routing interface definitions.
- tier4_planning_launch: remove redundant launch file.
- mission_planner: change mission planner to support AD API.

![planning-routing drawio](https://user-images.githubusercontent.com/43976882/182304917-550251d5-eefb-4182-8f20-b239eb48057b.png)

## Related links

https://github.com/autowarefoundation/autoware.universe/pull/1494
https://github.com/autowarefoundation/autoware_msgs/pull/33
https://github.com/autowarefoundation/autoware_adapi_msgs/pull/14

## Tests performed

1. Launch the planning simulation.
2. Check the routing state is UNSET.
3. Call the set_route_points API (refer to notes).
4. Check the routing state is SET.
5. Call the set_route_points API and check if it fails.
6. Start vehicle driving.
7. When vehicle goals, check the routing state is ARRIVED.
8. Call the clear_route API.
9. Call the set_route API (refer to notes)..

## Notes for reviewers

Sample Routes are from this position.
![Screenshot from 2022-08-02 16-20-43](https://user-images.githubusercontent.com/43976882/182321057-27b80c23-0dad-4807-aa8f-0aa45bbac2c9.png)

Sample1 (SetRoutePoints)
```bash
ros2 service call /planning/routing/set_route_points  autoware_ad_api_msgs/srv/SetRoutePoints "
header: {stamp: {sec: 1659424851, nanosec: 893215051}, frame_id: map}
goal:
  position: {x: 3835.50, y: 73765.91, z: 0.0}
  orientation: {x: 0.0, y: 0.0, z: -0.52530, w: 0.85091}"
```

Sample2 (SetRoutePoints)
```bash
ros2 service call /planning/routing/set_route_points  autoware_ad_api_msgs/srv/SetRoutePoints "
header: {stamp: {sec: 1659424851, nanosec: 893215051}, frame_id: map}
goal:
  position: {x: 3835.50, y: 73765.91, z: 0.0}
  orientation: {x: 0.0, y: 0.0, z: -0.52530, w: 0.85091}
waypoints:
 - position: {x: 3774.31, y: 73802.53, z: 0.0}
   orientation: {x: 0.0, y: 0.0, z: 0.22214, w: 0.97501}"
```

Sample1 (SetRoute)
```bash
ros2 service call /planning/routing/set_route  autoware_ad_api_msgs/srv/SetRoute "
header: {stamp: {sec: 1659424851, nanosec: 893215051}, frame_id: map}
goal:
  position: {x: 3835.50, y: 73765.91, z: 0.0}
  orientation: {x: 0.0, y: 0.0, z: -0.52530, w: 0.85091}
segments:
- {preferred: {id: 34976, type: lane}, alternatives: []}
- {preferred: {id: 34981, type: lane}, alternatives: []}
- {preferred: {id: 34585, type: lane}, alternatives: []}
- {preferred: {id: 34579, type: lane}, alternatives: []}
- {preferred: {id: 34780, type: lane}, alternatives: []}
- {preferred: {id: 34675, type: lane}, alternatives: []}"
```

Sample2 (SetRoute)
```bash
ros2 service call /planning/routing/set_route  autoware_ad_api_msgs/srv/SetRoute "
header: {stamp: {sec: 1659424851, nanosec: 893215051}, frame_id: map}
goal:
  position: {x: 3835.50, y: 73765.91, z: 0.0}
  orientation: {x: 0.0, y: 0.0, z: -0.52530, w: 0.85091}
segments:
- {preferred: {id: 34976, type: lane}, alternatives: []}
- {preferred: {id: 34981, type: lane}, alternatives: []}
- {preferred: {id: 34636, type: lane}, alternatives: []}
- {preferred: {id: 34564, type: lane}, alternatives: []}
- {preferred: {id: 34495, type: lane}, alternatives: []}
- {preferred: {id: 34507, type: lane}, alternatives: []}
- {preferred: {id: 34795, type: lane}, alternatives: []}
- {preferred: {id: 34606, type: lane}, alternatives: []}
- {preferred: {id: 34672, type: lane}, alternatives: []}
- {preferred: {id: 34675, type: lane}, alternatives: []}"
```

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.


## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.


## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.


After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/
